ipywidgets/interact.py

```diff
@@ -90,45 +117,47 @@ def _get_strrep(val):
     def __init__(self, function, **kwargs):
         # TODO: implement *args (difficult because of the name thing)
         # update names
+        # TODO: causes widget names to equal argument names,
```

Comment:
In the following lines, the current code renames the widgets with the corresponding variable name of the widget given in StaticInteract. This overrides user-defined names for the widgets and instead depicts the variable name. As a provisional workaround, I created a new optional argument in the widget base class: description, same name used in the official ipywidgets library. If no description is provided, then the displayed widget name is the variable name.